components/skeleton/skeleton-mixin.js

```diff
@@ -0,0 +1,100 @@
+import '../colors/colors.js';
+import { css } from 'lit-element/lit-element.js';
+
+export const skeletonStyles = css`
+	@keyframes loadingPulse {
+		0% { background-color: var(--d2l-color-sylvite); }
+		50% { background-color: var(--d2l-color-regolith); }
+		75% { background-color: var(--d2l-color-sylvite); }
+		100% { background-color: var(--d2l-color-sylvite); }
+	}
+	:host([skeleton]) .d2l-skeletize::before {
+		animation: loadingPulse 1.8s linear infinite;
+		background-color: var(--d2l-color-sylvite);
+		border-radius: 0.2rem;
+		bottom: 0;
+		content: '';
+		left: 0;
+		position: absolute;
+		right: 0;
+		top: 0;
+		z-index: 1;
+	}
+	:host([skeleton]) .d2l-skeletize {
+		border-color: transparent;
+		box-shadow: none;
+		color: transparent;
+		position: relative;
+	}
```

Comment:
One drawback to this approach is that you actually need a host element using the `SkeletonMixin` in order for the styles to get applied. This is fine for our core components, but if some app component just wanted to apply skeleton styles to a piece of typography (heading, link, etc), it's an extra step.

One solution to that could be to introduce another CSS class to force the skeleton ON, which would just add another selector to the styles, like this:

```css
:host([skeleton]) .d2l-skeletize,
.d2l-skeleton.d2l-skeletize {
   /* skeleton styles */
}
```

Then all a consumer would need to do would be `<h2 class="d2l-skeleton d2l-skeletize">` and then remove the `d2l-skeleton` class once things have loaded.

Thoughts? Can always add this later if it comes up.

Reply:
Seems reasonable/useful.  Would they remove `d2l-skeleton` or `d2l-skeletize`?  Given the selector, it looks like ti wouldn't matter which they remove... just wondering what the "most correct" would be.

Reply:
Yeah in this case adding/removing `d2l-skeleton` would be the most correct.

Reply:
I ended up not doing this for now -- it adds a lot of complexity and bulk to the CSS. We can look into adding it later if necessary.